sql: introduce pg_extension schema

otan opened the pull request:
Also added a barebones geography_columns, geometry_columns and
spatial_ref_sys table definition which I will delegate to another
reviewer - this bit is SQL Feature-y by itself :)

Release note (sql change): Introduced the pg_extension virtual schema,
which contains tables which are on the public schema when using postgres
style extensions.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/48754)
<!-- Reviewable:end -->

otan committed:
sql: introduce pg_extension schema

Also added a barebones geography_columns, geometry_columns and
spatial_ref_sys table definition which I will delegate to another
reviewer - this bit is SQL Feature-y by itself :)

Release note (sql change): Introduced the pg_extension virtual schema,
which contains tables which are on the public schema when using postgres
style extensions.

rohany commented on pkg/sql/logictest/testdata/logic_test/create_as:
```diff
@@ -132,7 +132,7 @@ query III colnames
 ----
 a  b  c
-1  2  4
+1  2  5
```
Why did this change?

otan replied:
foo is `CREATE TABLE foo (x, y, z) AS SELECT catalog_name, schema_name, sql_path FROM information_schema.schemata`, and we added a new schema.

rohany commented on pkg/sql/logictest/testdata/logic_test/geospatial:
```diff
@@ -103,7 +103,8 @@ INSERT INTO parse_test (geom, geog) VALUES
   (ST_GeomFromWKB(decode('0101000000000000000000F03F000000000000F03F', 'hex')), ST_GeogFromWKB(decode('0101000000000000000000F03F000000000000F03F', 'hex'))),
   (ST_GeomFromWKB(decode('0101000000000000000000F03F000000000000F03F', 'hex'), 3857), ST_GeogFromWKB(decode('0101000000000000000000F03F000000000000F03F', 'hex'), 3857)),
-  (ST_GeomFromEWKB(decode('0101000000000000000000F03F000000000000F03F', 'hex')), ST_GeogFromEWKB(decode('0101000000000000000000F03F000000000000F03F', 'hex')))
+  (ST_GeomFromEWKB(decode('0101000000000000000000F03F000000000000F03F', 'hex')), ST_GeogFromEWKB(decode('0101000000000000000000F03F000000000000F03F', 'hex'))),
```
are these changes part of adding pg_extension?

otan replied:
nah but it was an annoying rebase so i just moved this up from below.

otan commented:
bors r=rohany
